src/datachain/client/fsspec.py

```diff
@@ -89,9 +89,9 @@ def get_implementation(url: Union[str, os.PathLike[str]]) -> type["Client"]:
         from .local import FileClient
         from .s3 import ClientS3
 
-        protocol = urlparse(str(url)).scheme
+        protocol = urlparse(os.fspath(url)).scheme
```

Comment:
what is the difference, why do we need this?

Reply:
https://peps.python.org/pep-0519/

https://python-forum.io/thread-32670.html

Reply:
could you please summarize? is it a cosmetic change? linter complaining? any particular edge case it it solving?

Reply:
> This lack of support required users of pathlib to manually convert path objects to strings by calling str(path) which many found error-prone.

One issue in converting path objects to strings comes from the fact that the only generic way to get a string representation of the path was to pass the object to str().
This can pose a problem when done blindly as nearly all Python objects have some string representation whether they are a path or not,
e.g. str(None) will give a result that builtins.open() [5] will happily use to create a new file.

Reply:
okay, I see, thanks! Minor note: let's try to do this in a separate PR if possible.